core/trino-main/src/main/java/io/trino/server/security/Authentication.java

```diff
@@ -26,11 +27,11 @@
     private Authentication() {}
 
     /**
-     * When the user in the basic authentication header matches the x-trino-user header, we assume that the client does
-     * not want to force the runtime user name, and only wanted to communicate the authentication user.
+     * When principal name from authentication header matches the `x-trino-user` header, we assume that the client does
+     * not want to force the runtime username, and only wanted to communicate the authentication user.
      */
     public static void rewriteUserHeaderToMappedUser(
-            BasicAuthCredentials basicAuthCredentials,
+            Principal principal,
```

Comment:
Principal might be also something else than user name. Maybe simply pass a `String`?

Reply:
Fair point. Will adjust that.